### PR TITLE
[stable/redis] adding pod antiaffinity capability to redis slaves

### LINKED ISSUE #5249: [stable/redis] redis slaves pod antiaffinity
**Is this a request for help?**:

---

**Is this a BUG REPORT or FEATURE REQUEST?** (choose one):
FEATURE REQUEST

**Version of Helm and Kubernetes**:
Heml: 2.7.2
Kubernetes: v1.8.4

**Which chart**:
stable/redis

**What happened**:
N/A

**What you expected to happen**:
N/A

**How to reproduce it** (as minimally and precisely as possible):
N/A

**Anything else we need to know**:
N/A

### PR DESCRIPTION
**Pull request by RaymondArias**
**What this PR does / why we need it**:
Enable pod antiaffinity on Redis slaves. Helps ensure slaves are not concentrated on one node and be more resilient if one nodes crashed. 
**Which issue this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*: fixes #5249 

**Special notes for your reviewer**:

### CODE REVIEW

File: stable/redis/README.md [hunk 1]
```diff
@@ -131,6 +131,7 @@ The following table lists the configurable parameters of the Redis chart and the
 | `slave.securityContext.fsGroup`            | Group ID for the container (redis slave pod)                                                                   | `master.securityContext.fsGroup`                     |
 | `slave.securityContext.runAsUser`          | User ID for the container (redis slave pod)                                                                    | `master.securityContext.runAsUser`                   |
 | `slave.resources`                          | Redis slave CPU/Memory resource requests/limits                                                                | `master.resources`                                   |
+| `slave.podAntiAffinity`                          | Enable pod podAntiAffinity for slaves                                                                | `false`                                   |
```
Comment (timstoop) [inline stable/redis/README.md]: I'd prefer to see an `affinity` option instead, which the user can set themselves. This is what most other Charts seem to be moving to and it gives a user some more freedom. You could put your example (which should be good enough for 90% of all cases out there) in the README instead.

File: stable/redis/Chart.yaml [hunk 2]
```diff
@@ -1,5 +1,5 @@
 name: redis
-version: 3.0.5
+version: 3.0.6
```
Comment (timstoop) [inline stable/redis/Chart.yaml]: This adds a new feature, please bump the minor version instead of the patch level.

**Review by timstoop — CHANGES_REQUESTED** [review]

File: stable/redis/Chart.yaml (hunk 2 above)
Comment (RaymondArias) [reply stable/redis/Chart.yaml]: so version is 3.1.0?

File: stable/redis/README.md (hunk 1 above)
Comment (RaymondArias) [reply stable/redis/README.md]: Like the option to have a soft or hard affinity?

Comment (timstoop) [reply stable/redis/README.md]: Indeed, or to use different host labels. You can check `stable/mongodb-replicaset` for an example on how to do it in the Chart itself.

File: stable/redis/Chart.yaml (hunk 2 above)
Comment (timstoop) [reply stable/redis/Chart.yaml]: Indeed.

**Comment by RaymondArias** [comment]
Updated to affinity, follows similar logic to mongodb-replicaset

**Comment by timstoop** [comment]
@prydonius @sameersbn @tompizmor @carrodher What do you guys think? I think this is a worthwhile addition.

**Comment by RaymondArias** [comment]
Any update on this pr?

**Comment by carrodher** [comment]
Thanks for the contribution @RaymondArias, the tests fail because `3.1.0` version was released some days ago. Can you modify again `Chart.yaml`? Thanks

**Pull request merged**